src/hotspot/cpu/aarch64/gc/z/zBarrierSetAssembler_aarch64.cpp

```diff
@@ -680,17 +705,7 @@ void ZBarrierSetAssembler::copy_store_at(MacroAssembler* masm,
 
   bool is_dest_uninitialized = (decorators & IS_DEST_UNINITIALIZED) != 0;
 
-  if (is_dest_uninitialized) {
-    __ ldr(tmp1, Address(rthread, ZThreadLocalData::store_good_mask_offset()));
-    if (bytes == 8) {
-      __ bfi(src1, tmp1, 0, 16);
-    } else if (bytes == 16) {
-      __ bfi(src1, tmp1, 0, 16);
-      __ bfi(src2, tmp1, 0, 16);
-    } else {
-      ShouldNotReachHere();
-    }
-  } else {
+  if (!is_dest_uninitialized) {
```

Comment:
Why did you flip this condition so it's not consistent with the other copy_store_at?

Reply:
Ditto, too much like x86...

x86:

```
if (!dest_uninitialized) {
}

color source
store source
```

aarch64:

I added `copy_store_barrier_done` to color and store.

```
if (!is_dest_uninitialized) {
}

=>copy_store_barrier_done
```